tests/compartment_calls_inner.cc

```diff
@@ -10,6 +10,12 @@
 
 using namespace CHERI;
 
+extern "C" ErrorRecoveryBehaviour
+compartment_error_handler(ErrorState *frame, size_t mcause, size_t mtval)
+{
+	return ErrorRecoveryBehaviour::ForceUnwind;
+}
+
```

Comment:
This should be the default, do we need it to exist?

Reply:
I see, we don't unwind here because it's treated as a failure in the callee without the handler.  Please add a comment explaining this.

Reply:
Exactly. Great idea, I'm adding a comment and another inner compartment to tests both cases.

Reply:
It would also be good to update the error handling doc for some of these corner cases.  For example, how are each of the following handled:

 - Stack exhaustion
 - Trusted stack exhaustion
 - Invalid return addresses
 - Invalid callee addresses

We don't yet correctly handle the stack exhaustion case, but I'm finishing up the compiler change to generate the correct metadata.

Reply:
True, these corner cases are important and we have a good picture of how they should look like. I'll do that in a separate PR.